lib/Provider/OpenIDConnectClient.php

```diff
@@ -53,4 +53,21 @@ protected function startSession() {
     protected function commitSession() {
         $this->startSession();
     }
+    /**
+     * Gets the OIDC end session URL that will logout the user and redirect back to $post_logout_redirect_uri.
+     * 
+     * @param string $post_logout_redirect_uri Post signout redirect URL.
+     * 
+     * @return string The OIDC logout URL.
+     */
+    public function getEndSessionUrl($post_logout_redirect_uri)
+    {
+        $id_token_hint = $this->getIdToken();
+        $end_session_endpoint =  $this->getProviderConfigValue('end_session_endpoint');
+        $signout_params = array(
+            'id_token_hint' => $id_token_hint,
+            'post_logout_redirect_uri' => $post_logout_redirect_uri);
```

Comment:
If I set this line to `'post_logout_redirect_uri' => "https://my-nextcloud.domain.com");` it works, maybe we could create a new config variable, that makes it possible to redirect back to nextcloud and not only the oidc provider.
Right now the `oidc_login_logout_url `will be used, via `getEndSessionUrl($oidc_login_logout_url)`

Reply:
I believe that there is no need to add a new config variable. The existing config variables 'oidc_login_logout_url' and 'oidc_login_end_session_redirect' should do what you want to achieve. Could you please check the following code
https://github.com/pulsejet/nextcloud-oidc-login/blob/0d0d79b0bb49c9aed205a8666b8a7c9d3c14d07e/lib/Controller/LoginController.php#L126 which should make it clearer. 
You do however also need to configure the NextCloud client post-logout-redirect URL in the OIDC provider.

Then this should:

1. User initiates logout.
2. Nextcloud ends session.
3. Nextcloud redirect to OIDC provider end-session endpoint.
4. OIDC provider ends session.
5. OIDC provider redirect back to Nextcloud.

Reply:
> What is nextcloud config for oidc-login? Could you please list all config settings?

```
  'oidc_login_provider_url' => 'https://oidc.my-domain.com/realms/nextcloud',
  'oidc_login_client_id' => 'nextcloud',
  'oidc_login_client_secret' => 'secret',
  'oidc_login_end_session_redirect' => true,
  'oidc_login_logout_url' => 'https://oidc.my-domain.com/realms/nextcloud/protocol/openid-connect/logout?redirect_uri=https%3A%2F%2Fnextcloud.my-domain.com',
  'oidc_login_hide_password_form' => true,
  'oidc_login_redir_fallback' => false,
  'oidc_login_auto_redirect' => true,
  'oidc_login_button_text' => 'Login',
  'oidc_login_tls_verify' => true,
  'oidc_login_proxy_ldap' => true,
  'oidc_login_attributes' =>
  array (
    'id' => 'sub',
    'name' => '${given_name} ${family_name}',
    'mail' => 'email',
    'ldap_uid' => 'preferred_username',
  ),
  'oidc_login_scope' => 'openid profile',
  'oidc_login_disable_registration' => true,
  'oidc_create_groups' => false,
```
(removed configs not starting with `oidc`)
This is the current (that works), with having Keycloak 18 enabled with redirect_uri.

Reply:
The 'oidc_login_logout_url' should be 'https://nextcloud.my-domain.com/apps/oidc_login/oidc'. This will be the return URL to which the OIDC provider will redirect after logging out the user.

Reply:
> The 'oidc_login_logout_url' should be 'https://nextcloud.my-domain.com/apps/oidc_login/oidc'. This will be the return URL to which the OIDC provider will redirect after logging out the user.

Thanks, seems to work. 
Readme Update https://github.com/pulsejet/nextcloud-oidc-login/pull/174